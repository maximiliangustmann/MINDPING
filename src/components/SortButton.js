import React, { useState } from 'react'
import styled from 'styled-components'
import { ReactComponent as Sort } from '../img/sort.svg'
import PropTypes from 'prop-types'

SortButton.propTypes = {
  sortBySubcategory: PropTypes.func,
  sortByRating: PropTypes.func,
}

export default function SortButton({ sortBySubcategory, sortByRating }) {
  const [showMenu, setShowMenu] = useState(false)

  return (
    <SortButtonWrapper>
      <SortIcon onClick={() => setShowMenu(!showMenu)} data-cy="sort-icon" />
      {showMenu && (
        <ButtonWrapper>
          <Button
            onClick={() => sortBySubcategory()}
            data-cy="sort-subcategory"
          >
            Subcategory
          </Button>
          <Button onClick={() => sortByRating()} data-cy="sort-rating">
            Rating
          </Button>
        </ButtonWrapper>
      )}
    </SortButtonWrapper>
  )
}

const SortIcon = styled(Sort)`
  fill: var(--main-red);
  height: 40px;
  width: 40px;
  margin: 20px;
`

const SortButtonWrapper = styled.div`
  display: flex;
`

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const Button = styled.button`
  border: 2px solid var(--main-green);
  background: var(--main-white);
  border-radius: 20px;
  margin: 10px 5px;
  color: var(--main-black);
  font-weight: 300;
  font-size: 16px;
`
