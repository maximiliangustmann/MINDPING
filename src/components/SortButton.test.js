import React from 'react'
import renderer from 'react-test-renderer'
import SortButton from './SortButton'

describe('SortButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SortButton />)
    expect(tree).toMatchSnapshot()
  })
})
