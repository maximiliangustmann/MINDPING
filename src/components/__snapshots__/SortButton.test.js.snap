// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButton renders correctly 1`] = `
<div
  className="sc-AxirZ hlnYRa"
>
  <svg
    className="sc-AxjAm fZvzJc"
    data-cy="sort-icon"
    onClick={[Function]}
  >
    sort.svg
  </svg>
</div>
`;
